lib/LaTeXML/Package/pgfmath.code.tex.ltxml

```diff
@@ -93,29 +93,13 @@ DefMacro('\@@@show@pgfmatharg@{}', sub {
     print STDERR "MATH ARG $arg\n";
     return; });
 
-# A variant on Expanded that omits the outer {}
+# A plain argument that is expanded in the parameter type definition,
+# then used for pgf computations.
 DefParameterType('pgfNumber', sub {
     my ($gullet) = @_;
-    my $token;
-    do { $token = $gullet->readXToken(0);
-    } while (defined $token && $$token[1] == CC_SPACE);    # Inline ->getCatcode!
-
-    my $result = '';
-    if ($token->getCatcode == CC_BEGIN) {
-      my @tokens = ();
-      my $level  = 1;
-      while ($token = $gullet->readXToken(0)) {
-        my $cc = $$token[1];
-        if ($cc == CC_END) {
-          $level--;
-          last unless $level; }
-        elsif ($cc == CC_BEGIN) {
-          $level++; }
-        $result .= $$token[0]; } }
-    else {
-      $result = $token->getString; }
-    $result = "0" if $result eq '.';    # Apparently "." is a valid number!
-    return $result; });
+    my $pgf_number = ToString(Expand($gullet->readArg()));
```

Comment:
Shouldn't this be a `readXArg`?  (which we don't have, but probably should, essentially equivalent to the `Expanded` parameter type)

Reply:
Maybe? I am not quite sure. The uses I have seen are akin to:
```tex
\pgfmathcos@\pgfshade@angle
```

And the crucial aspect is that a single command sequence can be first read, then expanded, and its entire expansion is treated as the argument. I am not sure what `readXArg` would be specified to do, but it feels a bit more like `readArgX`, i.e. "first read in a plain, then expand it".

The `Expanded` parameter type is instead doing the TeX thing prescription of expanding from the get go, and taking the first non-expandable token as its argument, irrespective of how many expansion calls it had to do to get there. That's a difference to the pgf style of passing a plain argument around e.g. as in:
```tex
\def\pgfmathfpcos@#1{%
    \begingroup%
        \FPmul\pgfmathresult{#1}{0.017453292519943295}%
        \FPcos\pgfmathresult{\pgfmathresult}%
        \pgfmath@smuggleone\pgfmathresult%
    \endgroup%
}
```
```